### PR TITLE
爲RT1052移植QTMR的PWM

### PR DESCRIPTION
**Pull request by LeeChunHei**
## 拉取/合并请求描述：(PR description)

[
爲RT1052移植QTMR的PWM
Port QTMR to drv_pwm

已經在自製的imxrt1052 板子進行了對QTMR PWM驅動測試
Had tested QTMR PWM with a custom made imxrt1052 board
]

以下的内容不应该在提交PR时的message修改，修改下述message，PR会被直接关闭。请在提交PR后，浏览器查看PR并对以下检查项逐项check，没问题后逐条在页面上打钩。
The following content must not be changed in the submitted PR message. Otherwise, the PR will be closed immediately. After submitted PR, please use a web browser to visit PR, and check items one by one, and ticked them if no problem.

### 当前拉取/合并请求的状态 Intent for your PR

必须选择一项 Choose one (Mandatory):

- [ ] 本拉取/合并请求是一个草稿版本 This PR is for a code-review and is intended to get feedback
- [x] 本拉取/合并请求是一个成熟版本 This PR is mature, and ready to be integrated into the repo

### 代码质量 Code Quality：

我在这个拉取/合并请求中已经考虑了 As part of this pull request, I've considered the following:

- [x] 已经仔细查看过代码改动的对比 Already check the difference between PR and old code
- [x] 代码风格正确，包括缩进空格，命名及其他风格 Style guide is adhered to, including spacing, naming and other styles
- [x] 没有垃圾代码，代码尽量精简，不包含`#if 0`代码，不包含已经被注释了的代码 All redundant code is removed and cleaned up
- [x] 所有变更均有原因及合理的，并且不会影响到其他软件组件代码或BSP All modifications are justified and not affect other components or BSP
- [x] 对难懂代码均提供对应的注释 I've commented appropriately where code is tricky
- [x] 本拉取/合并请求代码是高质量的 Code in this PR is of high quality

### CODE REVIEW

File: bsp/imxrt/libraries/drivers/drv_pwm.c [hunk 1]
```diff
@@ -317,6 +324,224 @@ static rt_err_t imxrt_pwm4_init(PWM_Type *base)
...
+    }
+    else
+    {
+        rt_bool_t timer_is_on = base->CHANNEL[configuration->channel].CTRL & TMR_CTRL_CM_MASK;
+        rt_uint8_t duty = configuration->pulse * 100 / configuration->period;
+        QTMR_StopTimer(base, configuration->channel);
+        if (kStatus_Success != QTMR_SetupPwm(base, configuration->channel, 1000000000 / configuration->period, duty, DEFAULT_POLARITY, clk_freq))
+        {
+            LOG_E(LOG_TAG" setup pwm failed \n");
+            return RT_ERROR;
```
Comment (xinyigao) [inline bsp/imxrt/libraries/drivers/drv_pwm.c:395]: 返回错误添加负号，例：return -RT_ERROR;

File: bsp/imxrt/libraries/drivers/drv_pwm.c [hunk 2]
```diff
@@ -317,6 +324,224 @@ static rt_err_t imxrt_pwm4_init(PWM_Type *base)
...
+
+    qtmr_config.primarySource = (qtmr_primary_count_source_t)(psc + 8);
+    qtmr_clock_freq = CLOCK_GetFreq(kCLOCK_IpgClk) / (1 << psc);
+
+    QTMR_Init(base, channel, &qtmr_config);
+
+    if (kStatus_Success != QTMR_SetupPwm(base, channel, fre, duty, DEFAULT_POLARITY, qtmr_clock_freq))
+    {
+        LOG_E(LOG_TAG" setup pwm failed \n");
+        return RT_ERROR;
```
Comment (xinyigao) [inline bsp/imxrt/libraries/drivers/drv_pwm.c:439]: 返回错误添加负号，例：return -RT_ERROR;

**Review by xinyigao — CHANGES_REQUESTED** [review]
please modify :return -RT_ERROR;

File: bsp/imxrt/libraries/drivers/drv_pwm.c (hunk 2 above)
Comment (LeeChunHei) [reply bsp/imxrt/libraries/drivers/drv_pwm.c]: 已修改

File: bsp/imxrt/libraries/drivers/drv_pwm.c (hunk 1 above)
Comment (LeeChunHei) [reply bsp/imxrt/libraries/drivers/drv_pwm.c]: 已修改

**Comment by LeeChunHei** [comment]
> please modify :return -RT_ERROR;

已修改

**Review by xinyigao11 — APPROVED** [review]

**Pull request merged**